lib/data/wrapper.js

```diff
@@ -219,16 +230,23 @@ const data = {
 
     delete: (objectGetInfo, log, cb) => {
         const callback = cb || log.end;
-        // If objectGetInfo.key exists the md-model-version is 2 or greater.
-        // Otherwise, the objectGetInfo is just the key string.
-        const objGetInfo = (implName === 'sproxyd' || implName === 'cdmi') ?
-            objectGetInfo.key : objectGetInfo;
+        const isMdModelVersion2 = typeof(objectGetInfo) === 'string';
+        const isRequiredStringKey = constants.clientsRequireStringKey[implName];
+        const key = isMdModelVersion2 ? objectGetInfo : objectGetInfo.key;
+        const clientGetInfo = isRequiredStringKey ? key : objectGetInfo;
+
         log.trace('sending delete to datastore', {
-            implName,
-            key: objectGetInfo.key,
-            method: 'delete',
-        });
-        _retryDelete(objGetInfo, log, 0, err => {
+            implName, key, method: 'delete' });
+        // If the key is explicitly set to null, the part to
+        // be deleted doesn't really exist.
+        // This functionality is used by Scality-NFSD.
+        // Otherwise, the key is always defined
+        assert(key === null || key !== undefined);
+        if (key === null) {
+            callback(null);
```

Comment:
Do you need to send back `null` here? Otherwise go for `callback()` instead.

Reply:
because there are a lot of ```assert.strictEqual(err, null);``` here and there